### PR TITLE
Deduplicate curve sample methods for SW/TE

### PR DESCRIPTION
**Pull request by alexander-zw**
## Description

- There are no breaking changes, but ~now users can use their own implementation of `sample()`~ minor refactor to remove duplicate code. There should be no change in behavior or even performance.
- Simple documentation.
- I did not add unit tests because it's difficult to test randomness.

~closes: #308~

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (master)
- [x] Linked to GitHub issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] Wrote unit tests
- [x] Updated relevant documentation in the code
- [x] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the GitHub PR explorer

### CODE REVIEW

File: ec/src/models/mod.rs [hunk 1]
```diff
@@ -111,6 +113,24 @@ pub trait SWModelParameters: ModelParameters {
 
         res
     }
+
+    /// Generates a uniformly random affine instance of the group.
+    /// Implements `Distribution<GroupProjective<Self>>` for `Standard`.
+    #[inline]
```
Comment (alexander-zw) [inline ec/src/models/mod.rs:119]: I moved the inline directive to the method in the ModelParameters, but removed the ones in each group impl. Not sure if that's acceptable.

File: ec/src/models/short_weierstrass_jacobian.rs [hunk 2]
```diff
@@ -10,12 +15,6 @@ use ark_std::{
     vec::Vec,
 };
```
Comment (alexander-zw) [inline ec/src/models/short_weierstrass_jacobian.rs:17]: Not sure why there was a blank line here, I thought it made more sense to remove it, then ran cargo fmt.

**Review by mmagician — COMMENTED** [review]
What's the rationale of customising `sample`? From my understanding of [this issue](https://github.com/arkworks-rs/algebra/issues/308), we would want to allow downstream implementers of {SW/TE}ModelParameters to have a specialised implementation due to more efficient algorithms being known for some curves. I can't really see why we'd want this for `sample` - I'd have thought it'd mostly be used for testing and experimentation.

**Comment by alexander-zw** [comment]
@Pratyush ?

**Comment by alexander-zw** [comment]
Since we don't want to allow customization of sample(), I scaled down the changes to only make a small refactor.

**Comment by alexander-zw** [comment]
Never mind

**Pull request closed without merging**